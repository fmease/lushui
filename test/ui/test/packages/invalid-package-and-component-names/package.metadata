# TEST fail check

name: "???",
version: "0.0.0",

components: {
    "of": {
        type: executable,
        path: "",

        dependencies: {
            "0//": { component: "hey there", package: "-e" },
            "": { component: module, provider: distribution },
            data: {},
        },
    },
    # @Task add unquoted invalid name
},
