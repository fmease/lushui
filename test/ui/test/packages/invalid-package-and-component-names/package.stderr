error[E036]: the package name ‘???’ is not a valid word
  ┌─ ${TEST_FOLDER}/packages/invalid-package-and-component-names/package.metadata:3:8
  │
3 │ name: "???",
  │        ═══
  │

error[E036]: the component name ‘of’ is not a valid word
  ┌─ ${TEST_FOLDER}/packages/invalid-package-and-component-names/package.metadata:7:5
  │
7 │     "of": {
  │     ════
  │

error[E036]: the component name ‘0//’ is not a valid word
   ┌─ ${TEST_FOLDER}/packages/invalid-package-and-component-names/package.metadata:12:14
   │
12 │             "0//": { component: "hey there", package: "-e" },
   │              ═══
   │

error[E036]: the component name ‘hey there’ is not a valid word
   ┌─ ${TEST_FOLDER}/packages/invalid-package-and-component-names/package.metadata:12:34
   │
12 │             "0//": { component: "hey there", package: "-e" },
   │                                  ═════════
   │

error[E036]: the package name ‘-e’ is not a valid word
   ┌─ ${TEST_FOLDER}/packages/invalid-package-and-component-names/package.metadata:12:56
   │
12 │             "0//": { component: "hey there", package: "-e" },
   │                                                        ══
   │

error[E036]: the component name ‘’ is not a valid word
   ┌─ ${TEST_FOLDER}/packages/invalid-package-and-component-names/package.metadata:13:14
   │
13 │             "": { component: module, provider: distribution },
   │             ⟫⟪
   │

error[E036]: the component name ‘module’ is not a valid word
   ┌─ ${TEST_FOLDER}/packages/invalid-package-and-component-names/package.metadata:13:30
   │
13 │             "": { component: module, provider: distribution },
   │                              ══════
   │

error[E036]: the component name ‘data’ is not a valid word
   ┌─ ${TEST_FOLDER}/packages/invalid-package-and-component-names/package.metadata:14:13
   │
14 │             data: {},
   │             ════
   │

error: aborting due to 8 previous errors

